src/shared/resourcefetcher/httpResourceFetcher.ts

```diff
@@ -120,18 +119,25 @@ export class HttpResourceFetcher implements ResourceFetcher {
                 err ? reject(err) : resolve()
             })
 
-            timeout?.timer.catch(err => (pipe.end(), reject(err)))
+            timeout?.token.onCancellationRequested(({ type }) => {
+                getLogger().debug(`Download for "${this.logText()}" ${type === 'cancelled' ? type : 'timed out'}`)
+                pipe.destroy(new TimeoutError(type))
+            })
         })
 
-        return { requestStream, fsStream, done }
+        return Object.assign(done, { requestStream, fsStream })
```

Comment:
What benefit does the `Object.assign` give here? I feel like modifying objects at that level feels fairly odd...is there anything bad with the old impl?

Reply:
The main reasoning was that the promise is what you generally care about, not the streams. Probably will change it back, it's a small thing anyway.